### PR TITLE
add cartesian product executor

### PR DESCRIPTION
**Pull request by nevermore3**
as title

### CODE REVIEW

File: src/planner/Algo.h [hunk 1]
```diff
@@ -92,6 +92,47 @@ class ProduceAllPaths final : public SingleInputNode {
...
+        auto* varPtr = qctx_->symTable()->getVar(varName);
+        DCHECK(varPtr != nullptr);
+        inputVars_.emplace_back(varPtr);
+        for (const auto& name : varPtr->colNames) {
+            if (std::find(allColNames_.begin(), allColNames_.end(), name) != allColNames_.end()) {
+                return Status::SemanticError(
+                    "Var : %s , exist duplicate ColName : %s", varName.c_str(), name.c_str());
+            }
+            allColNames_.emplace_back(name);
+        }
```
Comment (CPWstatic) [inline src/planner/Algo.h:116]: Column name in `varPtr->colNames` might duplicate.

**Review by CPWstatic — COMMENTED** [review]
LGTM.

File: src/planner/Algo.h (hunk 1 above)
Comment (nevermore3) [reply src/planner/Algo.h:116]: will check it out in the for loop

**Review by CPWstatic — DISMISSED** [review]

File: src/planner/Algo.h [hunk 2]
```diff
@@ -92,6 +92,47 @@ class ProduceAllPaths final : public SingleInputNode {
...
+        return Status::OK();
+    }
+
+    const std::vector<std::string> inputVars() const {
```
Comment (yixinglu) [inline src/planner/Algo.h:120]: delete const for std::vector<std::string>

File: src/planner/Algo.h [hunk 3]
```diff
@@ -92,6 +92,47 @@ class ProduceAllPaths final : public SingleInputNode {
...
         : SingleInputNode(qctx, Kind::kProduceAllPaths, input) {}
 };
+
+class CartesianProduct final : public SingleDependencyNode {
+public:
+    static CartesianProduct* make(QueryContext* qctx, PlanNode* input) {
+        return qctx->objPool()->add(new CartesianProduct(qctx, input));
+    }
+
+    Status addVar(std::string varName) {
```
Comment (yixinglu) [inline src/planner/Algo.h:102]: please move the implementation of this function to cpp file, don't put all codes into header file.

File: src/planner/Algo.h (hunk 2 above)
Comment (nevermore3) [reply src/planner/Algo.h:120]: done

File: src/planner/Algo.h (hunk 3 above)
Comment (nevermore3) [reply src/planner/Algo.h:102]: done

**Review by CPWstatic — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**